scipy/stats/_stats.pyx

```diff
@@ -300,3 +301,170 @@ def _weightedrankedtau(ordered[:] x, ordered[:] y, intp_t[:] rank, weigher, bool
     tau = ((tot - (v + u - t)) - 2. * exchanges_weight[0]
            ) / np.sqrt(tot - u) / np.sqrt(tot - v)
     return min(1., max(-1., tau))
+
+
+# FROM MGCPY: https://github.com/neurodata/mgcpy
+
+# Distance transforms used for MGC and Dcorr
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+cdef _dense_rank_data(x):
+    cdef float64_t[:] x_view = x
+    _, v = np.unique(x_view, return_inverse=True)
+    return v + 1
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+cdef _rank_distance_matrix(ndarray distx):
+    # faster than np.apply_along_axis
+    cdef float64_t[:, :] distx_view = distx
+    return np.hstack([_dense_rank_data(distx_view[:, i]).reshape(-1, 1) for i in range(distx_view.shape[0])])
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+cdef _center_distance_matrix(ndarray distx, str global_corr='mgc', is_ranked=True):
+    cdef int n = distx.shape[0]
+    cdef ndarray rank_distx = np.zeros((<object> distx).shape)
```

Comment:
Does this cast do anything?

Reply:
I changed it to make this bit of code less confusing https://github.com/scipy/scipy/pull/10524/commits/fcfe4ea8a0c3eedfb76d7911ae4d95e986d2dda8